src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +265,42 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tockenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tockenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
-                && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                && !isEmptyBody(details.lcurly);
+    }
+
+    /**
+     * Checks that a right curly should be alone on a line or allows single-line format of block.
+     * ShouldBeAloneSLine = shouldBeAloneOrSingleLine it is for skip check "is longer than 100 char"
```

Comment:
Where did 100 characters come from? I don't see any mention of this in the original check.

`ShouldBeAloneSLine = shouldBeAloneOrSingleLine`
If you have to explain what your method name means, then you should just give the method a better name.